### DISCUSSION TITLE
eForms validation

### DISCUSSION THREAD

**Discussion by vilemvTS** [OP-TED/eForms-SDK#472] [Q&A] [answered]
Hi, Iam currently working on the eForms validation and I have a problem with converting schematron files into xslt and so far have been unable to do it. So I would like to ask how to do this or if there is xslt for eForms validation somewhere on github? Thanks in advance.

**Comment by rkottmann** [comment, marked as answer]
Agree that would be nice.

Until then: Have you tried schxslt ?

We use the schxslt XSLT to compile XSLT from the schematron like e.g. here https://projekte.kosit.org/eforms/validator-configuration-eforms/-/blob/main/build.xml#L247 

Schxslt has also other means like Java or command line among others: https://github.com/schxslt/schxslt#using-schxslt

Another more integrated option is to use https://github.com/phax/ph-schematron which is afaik used by TED CVS https://docs.ted.europa.eu/api/endpoints/cvs-ted-europa-eu.html in the background.

**Comment by Fynnyan** [comment]
We are currently using the "ph-schematron-pure" library to validate the our current xml notices.
It uses less memory and time.

Here a gist with some work in progress code and mvn config.
https://gist.github.com/Fynnyan/970ba2134df54582ada0ba4bb34f2a73

I found in a discussion, that this library is also used in the CVS - https://github.com/OP-TED/eForms-SDK/discussions/159#discussioncomment-3922803
